src/commands/add.py

```diff
@@ -0,0 +1,43 @@
+from src.shell import Shell
+from src.utils import Utils, Geometry, Crypt
+from src.password   import Password
+from getpass        import getpass
+
+class Add:
+
+    @staticmethod
+    def do(shell: Shell, line: str):
+        if shell.session.name is not None:
+            password = Password()
+            password.lab = Crypt.get_entry('label','  ')
+            password.url = Crypt.get_entry('url','    ')
+            password.com = Crypt.get_entry('comment', '')
+            password.usr = Crypt.get_entry('usr', '    ')
+            tentative = 2
+            while tentative != -1:
+                pas = getpass(' '*Geometry.INDENT+'pass    : ')
+                cof = getpass(' '*Geometry.INDENT+'confirm : ')
+                if (pas == cof):
+                    password.has = pas
+                    password.convert(shell.session.generate_path(), shell.session.key)
+                    break
+                else:
+                    if tentative == 0:
+                        Utils.print('code not added !')
+                        break
+                    else:
+                        Utils.print('!!! password do not match !!! left '+str(tentative)+' try')
+                    tentative -= 1
+            del password
+            shell.session.recover(Password)
+        else:
+            shell.help_add()
```

Comment:
Maybe to avoid confusion about the registration to the Shell class that happens elsewhere, the following equivalent would be better:
```
Add.help(shell)
```
Like this it is clear that the function called is the function defined in the `Add` class